File: php/src/Model/TeamRoster.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Result of replacing a team's membership wholesale.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TeamRoster
{
    public function __construct(
        public readonly string $teamId,
        /** @var list<string> */
        public readonly array $workerIds,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            teamId: (string) ($data['teamId'] ?? ''),
            workerIds: \array_values(\array_map(strval(...), $data['workerIds'] ?? [])),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TeamRoster.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * Result of replacing a team's membership wholesale.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TeamRoster {
    private String teamId;
    private List<String> workerIds;

    public String getTeamId() { return teamId; }
    public List<String> getWorkerIds() { return workerIds; }
}
